### PR TITLE
feat(Foxford): add presence

### PR DESCRIPTION
**Pull request by cramatsu**
## Description 
Adding `presence` for **[https://foxford.ru](https://foxford.ru)**

**Note to the founders of this project:**
This online school is the most visited in the Russian Federation (more than `300k` students).
Peace ✌🏿

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/C/Craiyon/dist/metadata.json [hunk 1]
```diff
@@ -14,5 +14,10 @@
...
 	"category": "other",
-	"tags": ["ai", "dalle", "image", "generation"]
-}
+	"tags": [
+		"ai",
+		"dalle",
+		"image",
+		"generation"
+	]
+}
```
Comment (plskz) [inline websites/C/Craiyon/dist/metadata.json:23]: revert

File: websites/H/Heroku/dist/metadata.json [hunk 2]
```diff
@@ -24,7 +24,11 @@
 	"thumbnail": "https://i.imgur.com/06zNE4U.png",
 	"color": "#211746",
 	"category": "other",
-	"tags": ["hosting", "application", "web"],
+	"tags": [
+		"hosting",
+		"application",
+		"web"
+	],
```
Comment (plskz) [inline websites/H/Heroku/dist/metadata.json:31]: revert

File: websites/R/Raven Code/dist/metadata.json [hunk 3]
```diff
@@ -20,4 +20,4 @@
 		"kod-paylaşım",
 		"raven-code"
 	]
-}
+}
```
Comment (plskz) [inline websites/R/Raven Code/dist/metadata.json:23]: revert

File: websites/S/Skype/dist/metadata.json [hunk 4]
```diff
@@ -42,4 +42,4 @@
 			"value": false
 		}
 	]
-}
+}
```
Comment (plskz) [inline websites/S/Skype/dist/metadata.json:45]: revert

File: websites/W/War Thunder/dist/metadata.json [hunk 5]
```diff
@@ -28,4 +28,4 @@
 		"airplane",
 		"battleship"
 	]
-}
+}
```
Comment (plskz) [inline websites/W/War Thunder/dist/metadata.json:31]: revert

Comment (cramatsu) [reply websites/W/War Thunder/dist/metadata.json:31]: Understood

**Review by plskz — CHANGES_REQUESTED** [review]
only commit your `websites/F/Foxfort` folder

**Comment by cramatsu** [comment]
> only commit your `websites/F/Foxfort` folder

Done

File: websites/F/Foxford/presence.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+		case pathHas("dashboard"):
+			presenceData.details = "Просматривает курсы";
+			break;
+		default:
+			presenceData.details = "Просматривает разделы сайта";
+			break;
+	}
+
+	if (presenceData.details) presence.setActivity(presenceData);
+	else presence.setActivity();
```
Comment (plskz) [inline websites/F/Foxford/presence.ts:72]: just a suggestion, same result

```suggestion
	presence.setActivity(presenceData);
```

**Review by plskz — COMMENTED** [review]
can you support more pages?

**Comment by cramatsu** [comment]
> can you support more pages? 

I can, I'll try to do it today.

File: websites/F/Foxford/presence.ts (hunk 6 above)
Comment (DarkVillager) [reply websites/F/Foxford/presence.ts:72]: Nah keept it as it is

Comment (cramatsu) [reply websites/F/Foxford/presence.ts:72]: > Nah keept it as it is

Okay, I won't touch these strings

**Comment by cramatsu** [comment]
Guys, look at my commits. About `foxford.ru`: I think the current paths are enough, over time I will update `foxford.ru`

**Comment by cramatsu** [comment]
I thought maybe we should split it into two prs?

**Comment by theusaf** [comment]
> I think maybe we should split this into two prs?

Yes, the two changes (adding Foxford and modifying jut.su) should be in two different PRs.

**Comment by cramatsu** [comment]
> > I think maybe we should split this into two prs?
> 
> Yes, the two changes (adding Foxford and modifying jut.su) should be in two different PRs.

So, I have undone all non-Foxford related changes.
Now, is PR ready to merge?

File: websites/F/Foxford/dist/metadata.json [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+	},
+	"service": "Foxford",
+	"description": {
+		"en": "Foxford is an online school for students in grades 1-11, teachers and parents.",
+		"ru": "Фоксфорд — онлайн-школа для учеников 1-11 классов, учителей и родителей."
+	},
+	"url": "foxford.ru",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/41WDw6H.jpeg",
+	"thumbnail": "https://i.imgur.com/41WDw6H.jpeg",
```
Comment (plskz) [inline websites/F/Foxford/dist/metadata.json:15]: The thumbnail **should** preferably be a [wide promotional card](https://i.imgur.com/3QfIc5v.jpg) or a [screenshot](https://i.imgur.com/OAcBmwW.png) if the first is **not** available.

File: websites/F/Foxford/dist/metadata.json [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+	"color": "#ff9300",
+	"category": "other",
+	"tags": ["school", "online-shool", "courses"]
```
Comment (plskz) [inline websites/F/Foxford/dist/metadata.json:18]:
```suggestion
	"tags": ["school", "learning", "courses"]
```

**Review by plskz — CHANGES_REQUESTED** [review]

File: websites/F/Foxford/dist/metadata.json (hunk 7 above)
Comment (cramatsu) [reply websites/F/Foxford/dist/metadata.json:15]: https://imgur.com/ZPIEalu
Is it ok?

Comment (cramatsu) [reply websites/F/Foxford/dist/metadata.json:15]: And another one that was made by me
https://imgur.com/5DZ70gr

File: websites/F/Foxford/dist/metadata.json (hunk 8 above)
Comment (cramatsu) [reply websites/F/Foxford/dist/metadata.json:18]: Thank you for noticing my typo

File: websites/F/Foxford/dist/metadata.json (hunk 7 above)
Comment (plskz) [reply websites/F/Foxford/dist/metadata.json:15]: use [imgur.com/ZEnNmek](https://i.imgur.com/ZEnNmek.png)

**Review by plskz — APPROVED** [review]

**Review by DarkVillager — APPROVED** [review]
Seems fine to me

**Review by RisingSunLight42 — COMMENTED** [review]
Maybe for every switch were you only edit the presence details you can create an object to store a pair of key/value correspond to the pathHas() and the details given to shorten the code.

**Comment by cramatsu** [comment]
> Maybe for every switch were you only edit the presence details you can create an object to store a pair of key/value correspond to the pathHas() and the details given to shorten the code.

Not a bad idea, now I'll rewrite the code for this

**Comment by cramatsu** [comment]
What do you think about this code?
Yes, I know that the DRY principle is violated in a couple of lines, 'cause it's raw

```ts
const presence = new Presence({
	clientId: "1019291803322679456",
});

function pathHas(pathName: string): boolean {
	return document.location.pathname.toLowerCase().includes(pathName);
}

const routes = {
	"conspects": "Просматривает конспект",
	"account": "Настраивает профиль",
	"externship": "Просматривает программу обучения",
	"calendar": "Смотрит календарь занятий",
	"daily-plan": "Смотрит план занятий",
	"objectives": "Просматривает задания",
	"progress": "Просматривает успеваемость",
	"dashboard": "Просматривает курсы",
} as const;

presence.on("UpdateData", async () => {
	const presenceData: PresenceData = {
		largeImageKey: "f1024",
	};

	switch (true) {
		case location.pathname.split("/").pop() in routes:
			presenceData.details =
				routes[
					location.pathname.split("/").pop() as unknown as keyof typeof routes
				];
			break;
		case pathHas("groups"):
			presenceData.details = "Просматривает урок";
			break;
		case pathHas("courses"):
			presenceData.state = document.querySelector(
				".Header_title__G-XGe"
			).textContent;
			presenceData.details = "Просматривает курс";
			break;
		case pathHas("users"):
			presenceData.details = "Просматривает профиль";
			presenceData.buttons = [
				{
					label: "Перейти",
					url: document.location.href,
				},
			];
			break;
		case pathHas("tasks"):
			presenceData.details = "Решает задачи";
			presenceData.buttons = [
				{
					label: "Перейти",
					url: document.location.href,
				},
			];
			break;
		default:
			presenceData.details = "Просматривает разделы сайта";
			break;
	}

	if (presenceData.details) presence.setActivity(presenceData);
	else presence.setActivity();
});
```

**About `group` route**
This route has the following structure: <protocol>://foxford.ru/groups/[0-9]

**Comment by RisingSunLight42** [comment]
Seems good for me

**Comment by cramatsu** [comment]
Is there any other way to improve the code or leave it as it is.
_I just ran out of ideas._

**Review by RisingSunLight42 — APPROVED** [review]

**Pull request closed without merging**